### PR TITLE
Entity: Do not replace `case class`

### PR DESCRIPTION
**Pull request by tindzk**

### CODE REVIEW

File: mapper/src/main/scala/org/scalarelational/mapper/Entity.scala [hunk 1]
```diff
@@ -7,17 +10,19 @@ import org.scalarelational.instruction.{Update, InsertSingle}
  * @author Matt Hicks <matt@outr.com>
  */
 trait Entity {
-  def toColumnValues: List[ColumnValue[Any]] =
-    throw new RuntimeException(s"@mapped annotation missing for table ${this.getClass}")
+  def mapTo[T <: Entity](table: Table): List[ColumnValue[Any]] =
+    macro mapped.mapTo[T]
+
+  def columns: List[ColumnValue[Any]]
```
Comment (darkfrog26) [inline mapper/src/main/scala/org/scalarelational/mapper/Entity.scala]: This is misleading in my opinion. ```Column``` is a very different thing from ```ColumnValue```.

File: mapper/src/main/scala/org/scalarelational/mapper/mapped.scala [hunk 2]
```diff
@@ -2,60 +2,40 @@ package org.scalarelational.mapper
...
+object mapped {
+  def simpleName(fullName: String) =
+    fullName.lastIndexOf('.') match {
+      case -1       => fullName
+      case position => fullName.substring(position + 1)
+    }
 
-object MappedMacro {
-  def impl(c: whitebox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
+  def mapTo[T](c: blackbox.Context)
```
Comment (darkfrog26) [inline mapper/src/main/scala/org/scalarelational/mapper/mapped.scala]: Why blackbox?

File: mapper/src/main/scala/org/scalarelational/mapper/Entity.scala (hunk 1 above)
Comment (tindzk) [inline mapper/src/main/scala/org/scalarelational/mapper/Entity.scala]: I agree. The reason is that it behaves like a DSL: ``def columns = mapTo[Supplier](Datastore.suppliers)``. How would you name it?

Comment (darkfrog26) [inline mapper/src/main/scala/org/scalarelational/mapper/Entity.scala]: I'm coming around on `columns` but I would probably name it `columnValues`

**Pull request merged**